flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/JobReconciler.java

```diff
@@ -109,6 +110,14 @@ public JobReconciler(
             }
         }
 
+        if (SavepointUtils.checkSavepointAnnotation(flinkApp)) {
+            triggerSavePoint(flinkApp, effectiveConfig);
+            SavepointUtils.removeSavepointAnnotation(flinkApp);
```

Comment:
Most probably you can enforce such a thing. We can introduce a counter based triggering mechanism later too, but that's less convenient for use IMO.

Reply:
I agree with removing the annotation after savepoint is done successfully. Then we will have a semantic that the savepoint is performing if the annotation exists.

Reply:
My question was related to whether modifying the CR may fail because the operator is normally only expected to modify the status field: https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#object-spec-and-status

I don't know what the convention for modifying metadata is though.

Reply:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

Reply:
I am not aware of any convention which prevents we could not remove the annotation. Correct me if I am wrong.

Reply:
@morhidi we should perhaps reconsider this approach of modifying the CR as it has further reaching implications. Essentially it makes the CR update non-idempotent, which can lead to operational problems. The same CR posted multiple times is not guaranteed to have the same outcome. And this is an issue as the source of the CR would normally be managed in an external system (git repo for example) and applied through a CD pipeline.

Reply:
@tweise what would be the alternative solution?

Reply:
I think the concern make sense, since the operator's semantic is try to reconcile to the desired state, and the desired state changed by the operator itself is not common way.
Can we attach the annotation of  `flinkdeployments.flink.apache.org/trigger-savepoint-timestamp=1646411987`
and stored the last triggered timestamp in status. If the status timestamp < the spec timestamp , It will need to trigger again and update the status.

Reply:
How about just not removing the annotation?

Reply:
How about an explicit field `triggerSavepointNonce` that can be modified whenever a trigger is desired. That would work naturally with reconcile semantics.

Reply:
I introduced a new field savepointGeneration in the CR @tweise that serves as a counter